arch/xtensa/Kconfig

```diff
@@ -82,6 +82,7 @@ config ARCH_CHIP_ESP32S3
 	select ARCH_HAVE_RESET
 	select ARCH_HAVE_TESTSET
 	select ARCH_VECNOTIRQ
+	select FORCE_LIBC_ARCH_KERNEL if BUILD_PROTECTED
 	select LIBC_ARCH_MEMCPY if BUILD_FLAT
```

Comment:
@tmedicci I can't fully understand your change. From my understanding, ARCH_CHIP_ESP32S3 only enable the assembly version when BUILD_FLAT is enabled, and BUILD_PROTECTED already use the C version which already match your expectation. why need add FORCE_LIBC_ARCH_KERNEL?

Reply:
The option `FORCE_LIBC_ARCH_KERNEL` forces the kernel to use the alternative version (or ROM functions, in this case) on the kernel while allowing the userspace to build the C version.

Specifically, S3 does not allow userspace to directly access the ROM functions through the MMU. That's why the userspace would need to build the C version. On the other hand, the Wi-Fi driver requires the usage of the ROM functions on the kernel-side.

Reply:
should we change FORCE_LIBC_ARCH_KERNEL  to LIBC_ARCH_FORCE_KERNEL?

Reply:
It makes sense. Just a minute!

Reply:
done! I also updated the `help` function to better explain this flag usage.